master/buildbot/db/logs.py

```diff
@@ -345,6 +345,40 @@ def thdcompressLog(conn):
         saved = yield self.db.pool.do(thdcompressLog)
         defer.returnValue(saved)
 
+    def deleteOldLogChunks(self, older_than_timestamp):
+        def thddeleteOldLogs(conn):
+            model = self.db.model
+            res = conn.execute(sa.select([sa.func.count(model.logchunks.c.content)]))
+            count1 = res.fetchone()[0]
+            res.close()
+
+            # update log types older than timestamps
+            res = conn.execute(
+                model.logs.update()
+                .where(model.logs.c.stepid.in_(
+                    sa.select([model.steps.c.id])
+                    .where(model.steps.c.started_at < older_than_timestamp)))
+                .values(type='d')
+            )
+            res.close()
+
+            # query all logs older than timestamps
+            q = sa.select([model.logs.c.id])
+            q = q.select_from(model.logs.join(model.steps, model.steps.c.id == model.logs.c.stepid))
+            q = q.where(model.steps.c.started_at < older_than_timestamp)
+
+            # delete their logchunks
+            res = conn.execute(
+                model.logchunks.delete()
+                .where(model.logchunks.c.logid.in_(q))
```

Comment:
I did the test on copy of nine.bb.net (sqlite3) database with 419982 logchunks.
deleting all logchunks
```
DELETE FROM logchunks WHERE logchunks.logid IN (SELECT logs.id
   ...>         FROM logs JOIN steps ON steps.id = logs.stepid
   ...>         WHERE steps.started_at < 1493027316)
   ...> ;
sqlite> vacuum;
```
it takes ~10s on first run, and <1s on next run
db goes from 1.3G to 279M

I think this good enough numbers to avoid doing any more optimization